samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -26,7 +26,7 @@
 
 public abstract class AbstractApplicationRunner implements ApplicationRunner {
```

Comment:
I have a general question, not necessarily related to this PR. JDK8 allows interface methods to have default implementations. What is the standard criteria to decide when the default implementation belongs to the interface and when to put it in an abstract class?

The reason I ask is that AbstractApplicationRunner provides an implementation of streamFromConfig. Does that mean we allow classes derived from AbstractApplicationRunner to override this method? Is there a use-case for this? 
Nothing stops the user from directly implementing the ApplicationRunner interface or extending AbstractApplicationRunner. So, I am trying to understand the motivation behind this choice. 

_Clearly, I didn't review the PR that committed this changed. So sorry about asking questions this late_